Make tab-nav a bit more responsive

tomalec opened the pull request:
### Changes proposed in this Pull Request:

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

This PR makes the tab navigation a bit more responsive, my making tabs wrap when the container is too narrow. It also adds the  bottom "bar" to the items, sot when the elements wrap, the "bar" is visible under all tabs.

Closes #2473

_Replace this with a good description of your changes & reasoning._


### Screenshots:

#### Before
![image](https://github.com/user-attachments/assets/7e93645a-5aac-44a3-b645-78720d4bb796)

#### After
![image](https://github.com/user-attachments/assets/d1119328-238b-4b95-bd1d-7f95a0876776)

<!--- Optional --->


### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1.  Go to G4W dashboard `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard`
2. Change the viewport size to something narrower
3. Check that all the tabs are visible
4. Go to Product feed `/wp-admin/admin.php?page=wc-admin&issueType=product&path=%2Fgoogle%2Fproduct-feed`
5. Check that all "Issues to resolve" tabs are visible.


### Additional details:

1. We may consider implementing more dedicated UI for the narrower screens, but I believe it's a good enough quick fix in the meantime.


<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with change type prefix`(Fix|Add|…) - `, for example:
> Break - A change breaking previous API or functionality.
> Add - A new feature, function or functionality was added.
> Update - Big changes to something that wasn't broken.
> Fix - Took care of something that wasn't working.
> Tweak - Small change, that isn't actually very important.
> Dev - Developer-facing only change.
> Doc - Updated customer or developer facing documentation

If you remove the "Changelog entry" header, the Pull Request title will be used as the changelog entry.

Add the `changelog: none` label if no changelog entry is needed.
-->
### Changelog entry

> Fix - Make the tab navigation tabs wrap when the screen narrows.

tomalec committed:
Make tab-nav a bit more responsive

Fixes https://github.com/woocommerce/google-listings-and-ads/issues/2473

tomalec committed:
Remove padding from issues table nav on mobile

tomalec committed:
Remove duplicated box-shadow entry

eason9487 commented on js/src/components/app-tab-nav/index.scss:
```diff
@@ -1,13 +1,14 @@
 		background: transparent;
 		border: none;
-		box-shadow: none;
+		box-shadow: inset 0 -1px 0 #ccc;
```
Suggest using the same color variable as line 5 instead of using `#ccc`.

eason9487 reviewed (approved):
Thanks for the fix. LGTM.

The only comment should not need another round of code reviews.

tomalec committed:
Use color code from variable

Address https://github.com/woocommerce/google-listings-and-ads/pull/2704#discussion_r1865244457
